chats/apps/api/v1/rooms/viewsets.py

```diff
@@ -214,16 +213,47 @@ def perform_update(self, serializer):
         msg.notify_room("create")
 
         if old_user is None and user:  # queued > agent
-            instance.notify_queue("update", transferred_by=self.request.user.email)
+            instance.notify_queue("update")
         elif old_user is not None:
-            instance.notify_user(
-                "update", user=old_user, transferred_by=self.request.user.email
-            )
+            instance.notify_user("update", user=old_user)
             if queue:  # agent > queue
-                instance.notify_queue("update", transferred_by=self.request.user.email)
+                instance.notify_queue("update")
             else:  # agent > agent
-                instance.notify_user("update", transferred_by=self.request.user.email)
+                instance.notify_user("update")
 
     def perform_destroy(self, instance):
         instance.notify_room("destroy", callback=True)
         super().perform_destroy(instance)
+
+    @action(
+        detail=True,
+        methods=["PATCH"],
+    )
+    def update_custom_fields(self, request, pk=None):
+        custom_fields_update = request.data
+        data = {"fields": custom_fields_update}
+
+        if pk is None:
+            return Response(
+                {"Detail": "No room on the request"}, status.HTTP_400_BAD_REQUEST
+            )
+        elif not custom_fields_update:
+            return Response(
+                {"Detail": "No custom fields on the request"},
+                status.HTTP_400_BAD_REQUEST,
+            )
+
+        room = get_editable_custom_fields_room({"uuid": pk, "is_active": "True"})
+
+        update_flows_custom_fields(
+            project=room.queue.sector.project,
+            data=data,
+            contact_id=room.contact.external_id,
+        )
+
+        update_custom_fields(room, custom_fields_update)
```

Comment:
Call `update_custom_fields` before `update_flows_custom_fields`
imagine the situations:

### Use Case 1 (As it currently stands)
you call the flows, it makes the changes successfully but when it comes to persisting it in the Chats database, an error occurs, the format of the custom fields is wrong.

To reverse this you would need to call Flows reversing the data correctly.

### Use case 2 (calling `update_custom_fields` before `update_flows_custom_fields`)
If an error occurs during data persistence, you prevent Flows from being called and if there is a problem during the Flows call, it is much easier and safer to revert directly in Chats itself

This way it is easier to maintain consistency between Chats and Flows.